src/main/groovy/io/openliberty/tools/gradle/utils/LooseEarApplication.groovy

```diff
@@ -1,42 +1,50 @@
 package io.openliberty.tools.gradle.utils
 
-import java.io.File
+import io.openliberty.tools.common.plugins.config.LooseApplication
+import io.openliberty.tools.common.plugins.config.LooseConfigData
+import org.apache.commons.io.FilenameUtils
 import org.gradle.api.Project
-import org.gradle.plugins.ear.EarPluginConvention
 import org.gradle.api.Task
+import org.gradle.api.logging.Logger
 import org.gradle.plugins.ear.Ear
-import org.gradle.api.internal.file.FileResolver
-import org.gradle.api.artifacts.Dependency
-import org.w3c.dom.Element;
-import org.apache.commons.io.FilenameUtils
-
-import io.openliberty.tools.common.plugins.config.LooseConfigData
-import io.openliberty.tools.common.plugins.util.PluginExecutionException
-import io.openliberty.tools.common.plugins.config.LooseApplication
+import org.w3c.dom.Element
 
 public class LooseEarApplication extends LooseApplication {
     
     protected Task task;
+    protected Logger logger;
 
-    public LooseEarApplication(Task task, LooseConfigData config) {
-        super(task.getProject().getBuildDir().getAbsolutePath(), config)
+    public LooseEarApplication(Task task, LooseConfigData config, Logger logger) {
+        super(task.getProject().getLayout().getBuildDirectory().getAsFile().get().getAbsolutePath(), config)
         this.task = task
+        this.logger = logger
     }
 
     public void addSourceDir() throws Exception {
-        File sourceDir = new File(task.getProject().path.replace(":","") + "/" + task.getProject().getConvention().getPlugin(EarPluginConvention.class).getAppDirName())
-        config.addDir(sourceDir, "/")
+        if (task.getProject().getPlugins().hasPlugin("ear")) {
+            Ear ear = (Ear) task.getProject().ear
+            File sourceDir = new File(task.getProject().path.replace(":","") + "/" + ear.getAppDirectory().getAsFile().get().getAbsolutePath())
```

Comment:
What does `getAppDirectory()` return? Before we were getting just a directory name, not a full path. This seems like it is getting a full path, which when concatenated onto the project path would not be right.